### ISSUE TITLE
Exahustive exception capture in mongoBackend

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#136]
*(Original name of this issue was: "Disable assertions in production?", but is has evolved during the discussion to the current name)*

It seems that mongo driver code includes assertions. Sometimes, we have got messages such as the following one:

```
Thu Oct 17 16:54:08 Assertion: 13111:wrong type for field (value) 16 != 4
terminate called after throwing an instance of 'mongo::MsgAssertionException'
  what():  wrong type for field (value) 16 != 4
```

Is it a good idea that code installed in production may abort due to an assertion fail? Should we compile using NDEBUG to avoid this behavior in production?

**Effort: 4 man day**

**Comment by fgalan** [comment]
Partially addressed in PR #1394. What is pending after that PR:

* Check that all the code is *not* using directly BSONObj get methos *after* landing the "new cache"
* Stop using STR_FIELD and C_STR_FIELD (as now the syntax is completely aligned with the function to get the field data).

**Comment by fgalan** [comment]
Implementation finished in PR https://github.com/telefonicaid/fiware-orion/pull/1416
